### PR TITLE
[TT-10477, TT-10633] Add GraphQLIntrospectionConfig struct

### PR DESCRIPTION
**Pull request by buraksezer**
A new configuration element has been added. `GraphQLIntrospectionConfig` struct is a field of `GraphQLConfig` struct. 

```
"introspection": {
        "disabled": false
}
```
See [TT-10477](https://tyktech.atlassian.net/browse/TT-10477) for details. 

Related PR https://github.com/TykTechnologies/tyk/pull/5792

[TT-10477]: https://tyktech.atlassian.net/browse/TT-10477?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by buraksezer** [comment]
Hey @buraksekili and @komalsukhani, we want to add a new field to the API definitions. I added `GraphQLIntrospectionConfig` to the `GraphQLConfig` struct. How can we proceed from this point? Do we need to update any other structs or tests to complete this task? I saw an auto-generated file, but I don't know how to run the generator.

**Comment by buraksekili** [comment]
thank you @buraksezer,

After updating a Go struct used in the generation of Kubernetes manifests, it is advisable to execute the following command:

```bash
make generate manifests helm install
```

> for reference: [Useful Make Commands](https://github.com/TykTechnologies/tyk-operator/blob/master/docs/development.md#useful-make-commands).

As for the next steps, we need to incorporate testing of this pull request into our standard QA process. However, our current focus is on the upcoming release of the Helm chart. Also, it'd be great to check the possibility of integrating this feature with relevant Kubernetes features, such as secret management. If there is no integration possible, we can merge it directly after testing.
 
If the pull request is not urgent, let's keep it open until the Helm release is completed. This will allow us to address the PR in a more comprehensive manner.

File: api/model/api.go [hunk 1]
```diff
@@ -1163,6 +1163,13 @@ type GraphQLConfig struct {
 
 	// Supergraph holds the configuration for a GraphQL federation supergraph.
 	Supergraph GraphQLSupergraphConfig `json:"supergraph,omitempty"`
+
+	// Introspection holds the configuration for GraphQL Introspection
+	Introspection GraphQLIntrospectionConfig `json:"introspection"`
```
Comment (buraksekili) [inline api/model/api.go:1168]: Is `Introspection` required field in GraphQLConfig? If not, we should add the `omitempty` tag to make it optional in CRD.

The same thing also applies to `Disabled` field in `GraphQLIntrospectionConfig`.

**Review by buraksekili — COMMENTED** [review]
thanks @buraksezer, lgtm overall with a minor comment. can you also please update CHANGELOG according to your changes?

**Comment by buraksezer** [comment]
> thanks @buraksezer, lgtm overall with a minor comment. can you also please update CHANGELOG according to your changes?

Hey @buraksekili, I completed the requested changes and additions. Could you please review the PR again?

**Review by buraksekili — APPROVED** [review]
lgtm, thank you @buraksezer for the contribution. there is a minor conflict in the CHANGELOG though. Can you please resolve it if you have availability? If not, please let me know.

**Comment by buraksezer** [comment]
> lgtm, thank you @buraksezer for the contribution. there is a minor conflict in the CHANGELOG though. Can you please resolve it if you have availability? If not, please let me know. Also, can you please update the k8s manifests as mentioned [here](https://github.com/TykTechnologies/tyk-operator/pull/662#issuecomment-1822637085)?

Hey @buraksekili, I have fixed the conflict.

**Pull request merged**